### ISSUE TITLE
location of useLocation does not re-render on queryString pushstate

### ISSUE THREAD

**Issue by bdombro** [molefrog/wouter#168]
If one navigates from '/' to '/?hello=world', components using useLocation do not re-render.

Recommended Fix:
In ./use-location.js, make the events searchQuery aware by  replacing `currentPathname` with

```javascript
const currentPath = (base: string) => {
    const {pathname, search} = location
    const path = !pathname.toLowerCase().indexOf(base.toLowerCase())
        ? pathname.slice(base.length) || "/"
        : "~" + pathname;
    return path + search
} 
```

**Comment by cbbfcd** [comment]
@bdombro  welcome PR. 😄

**Comment by molefrog** [comment]
Thanks for reporting. If we patch the `currentPathname` with the code proposed, it might break the existing apps (since now the location will return `"/?hello=world"` and not `"/"`). I guess we could patch https://github.com/molefrog/wouter/blob/master/use-location.js#L20 instead.

**Comment by molefrog** [comment]
Released the fix in v2.7.2, thanks @cbbfcd
